pkg/api/testing/unstructured_test.go

```diff
@@ -130,6 +133,49 @@ func TestRoundTrip(t *testing.T) {
 	}
 }
 
+func TestRoundTripWithEmptyCreationTimestamp(t *testing.T) {
+	for groupKey, group := range testapi.Groups {
+		for kind := range group.ExternalTypes() {
+			if nonRoundTrippableTypes.Has(kind) {
+				continue
+			}
+			item, err := api.Scheme.New(group.GroupVersion().WithKind(kind))
+			if err != nil {
+				t.Fatalf("Couldn't create external object %v: %v", kind, err)
+			}
+			t.Logf("Testing: %v in %v", kind, groupKey)
+
+			unstrBody, err := unstructured.DefaultConverter.ToUnstructured(item)
+			if err != nil {
+				t.Errorf("ToUnstructured failed: %v", err)
+				return
+			}
+
+			unstructObj := &metaunstruct.Unstructured{}
+			unstructObj.Object = unstrBody
+
+			if meta, err := meta.Accessor(unstructObj); err == nil {
+				meta.SetCreationTimestamp(metav1.Time{})
+			}
+
+			_, ok := unstructObj.Object["metadata"]
+			if !ok {
+				t.Logf("No metadata field found for %v, Skipping...", kind)
```

Comment:
If the item has a meta accessor and the Unstructured is missing a `metadata` key, does that seem like an erroneous condition?